test/integration/protocol_integration_test.h

```diff
@@ -18,6 +18,27 @@ class ProtocolIntegrationTest : public HttpProtocolIntegrationTest {
              << "Please use DownstreamProtocolIntegrationTest instead of ProtocolIntegrationTest.";
     }
   }
+
+  void expectWireBytesSentAndReceived(std::string log_name, int log_index, int h1_wire_bytes_sent,
+                                      int h1_wire_bytes_received, int h2_wire_bytes_sent,
+                                      int h2_wire_bytes_received) {
+    auto integer_near = [](int x, int y) -> bool { return std::abs(x - y) <= (x / 20); };
+    std::string access_log = waitForAccessLog(log_name, log_index);
+    std::vector<std::string> log_entries = absl::StrSplit(access_log, ' ');
+    int UPSTREAM_WIRE_BYTES_SENT = std::stoi(log_entries[0]),
+        UPSTREAM_WIRE_BYTES_RECEIVED = std::stoi(log_entries[1]);
+    if (upstreamProtocol() == Http::CodecType::HTTP1) {
+      EXPECT_EQ(h1_wire_bytes_sent, UPSTREAM_WIRE_BYTES_SENT);
+      EXPECT_EQ(h1_wire_bytes_received, UPSTREAM_WIRE_BYTES_RECEIVED);
+    }
+    if (upstreamProtocol() == Http::CodecType::HTTP2) {
+      // Because of non-deterministic h2 compression, the same plain text length don't map to the
+      // same number of wire bytes.
+      EXPECT_TRUE(integer_near(h2_wire_bytes_sent, UPSTREAM_WIRE_BYTES_SENT));
```

Comment:
It looks like right now the test always gets 0 from access log, so some plumbing got broken. Please also print the values in the test result like so to make it easy to understand why the test failed:
`EXPECT_TRUE(integer_near(h2_wire_bytes_sent, UPSTREAM_WIRE_BYTES_SENT)) << "Expected: " << h2_wire_bytes_sent << " actual: " << UPSTREAM_WIRE_BYTES_SENT;

Reply:
makes sense.